ldi-orchestrator/ldio-connectors/ldio-http-in-poller/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/HttpInputPoller.java

```diff
@@ -7,42 +7,52 @@
 import be.vlaanderen.informatievlaanderen.ldes.ldi.requestexecutor.valueobjects.Response;
 import be.vlaanderen.informatievlaanderen.ldes.ldi.services.ComponentExecutor;
 import be.vlaanderen.informatievlaanderen.ldes.ldi.types.LdiAdapter;
+import be.vlaanderen.informatievlaanderen.ldes.ldio.config.PollingInterval;
 import be.vlaanderen.informatievlaanderen.ldes.ldio.exceptions.MissingHeaderException;
 import be.vlaanderen.informatievlaanderen.ldes.ldio.exceptions.UnsuccesfulPollingException;
 import be.vlaanderen.informatievlaanderen.ldes.ldio.types.LdioInput;
 import io.micrometer.observation.ObservationRegistry;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.http.HttpStatusCode;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
 
+import java.time.Instant;
 import java.util.List;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
 
-public class HttpInputPoller extends LdioInput {
+import static be.vlaanderen.informatievlaanderen.ldes.ldio.config.PollingInterval.TYPE.CRON;
+
+public class HttpInputPoller extends LdioInput implements Runnable {
 	public static final String NAME = "be.vlaanderen.informatievlaanderen.ldes.ldio.LdioHttpInPoller";
-	private final ScheduledExecutorService scheduler;
+	private final ThreadPoolTaskScheduler scheduler;
 	private final RequestExecutor requestExecutor;
 	private final List<? extends Request> requests;
 	private final boolean continueOnFail;
 	private static final Logger log = LoggerFactory.getLogger(HttpInputPoller.class);
 	private static final String CONTENT_TYPE = "Content-Type";
 
 	public HttpInputPoller(String pipelineName, ComponentExecutor executor, LdiAdapter adapter, ObservationRegistry observationRegistry, List<String> endpoints,
-						   boolean continueOnFail, RequestExecutor requestExecutor) {
+	                       boolean continueOnFail, RequestExecutor requestExecutor) {
 		super(NAME, pipelineName, executor, adapter, observationRegistry);
 		this.requestExecutor = requestExecutor;
 		this.requests = endpoints.stream().map(endpoint -> new GetRequest(endpoint, RequestHeaders.empty())).toList();
 		this.continueOnFail = continueOnFail;
-		this.scheduler = Executors.newSingleThreadScheduledExecutor();
+		this.scheduler = new ThreadPoolTaskScheduler();
+		this.scheduler.setErrorHandler(t -> {
```

Comment:
is this supposed to be empty? Maybe add a comment to explain the purpose of this?